lib/elixir/lib/module/types/pattern.ex

```diff
@@ -332,6 +337,10 @@ defmodule Module.Types.Pattern do
   Refines the type variables in the typing context using type check guards
   such as `is_integer/1`.
   """
+  def of_guard(expr, %{context: stack_context} = stack, context) when stack_context != :pattern do
+    of_guard(expr, %{stack | context: :pattern}, context)
```

Comment:
If you want to be consistent with the compiler, we call it `:match` instead of `:pattern`  (and `:guard` for guards).

Reply:
Good point. Inference behaves the same for matches and guards so two different names is not needed. Should we keep using `:pattern` or chose one of `:match` or `:guard`?

Reply:
Ah, so `:pattern` is fine!